include/deal.II/matrix_free/operators.h

```diff
@@ -328,6 +328,75 @@ namespace MatrixFreeOperators
 
 
 
+  /**
+   * This class implements the operation of the action of a Laplace matrix.
+   *
+   * @author Denis Davydov, 2016
+   */
+  template <int dim, int fe_degree, int n_q_points_1d = fe_degree+1, int n_components = 1, typename Number = double>
+  class LaplaceOperator : public Base<dim, Number>
+  {
+  public:
+
+    /**
+     * Constructor.
+     */
+    LaplaceOperator ();
+
+    /**
+     * The diagonal is approximated by computing a local diagonal matrix per element
+     * and distributing it to the global diagonal. This will lead to a slightly
+     * different diagonal for the case with hanging nodes.
```

Comment:
In the step-37 tutorial that I'm currently rewriting (I hope to finish a first version today, but I had too many students in my office today so I got less done than expected... ;-)) I used the following rather longish text (that text still needs to go through the review I admit):

```
  // Note that the code that calls distribute_local_to_global on the vector to
  // accumulate the diagonal entries into the global matrix needs to be
  // handled with care. For operators with hanging node constraints that
  // distribute an integral contribution of a constrained DoF to several other
  // entries inside the distribute_local_to_global call, the vector interface
  // used here does not exactly compute the diagonal entries. The result is
  // correct up to discretization accuracy as shown in <a
  // href="http://dx.doi.org/10.4208/cicp.101214.021015a">Kormann
  // (2016), section 5.3</a>, but not equal.
```